examples/aggregate_merging/src/aggregates.rs

```diff
@@ -0,0 +1,129 @@
+impl<
+        E: Send
+            + Sync
+            + Serialize
+            + DeserializeOwned
+            + // EventStore bounds
+            Into<ProjectorEvent>
+            + Clone, // for the CounterProjector
+    > Aggregate<E>
+where
+    Aggregate<E>: Identifier,
+{
+    pub async fn new(pool: &Pool<Sqlite>) -> Result<Self, CounterError> {
+        Ok(Self {
+            event_store: Self::new_store(pool).await?,
+        })
+    }
+
+    pub async fn new_store(pool: &Pool<Sqlite>) -> Result<Store<E>, CounterError> {
+        // Any aggregate based off this template will project to the CounterProjector
+        let projectors: Vec<Box<dyn SqliteProjector<E, CounterError> + Send + Sync>> = vec![Box::new(CounterProjector)];
+
+        SqliteStore::new::<Self>(pool, projectors, vec![]).await
+    }
+}
+
+pub type AggregateA = Aggregate<EventA>;
+pub type AggregateB = Aggregate<EventB>;
+
+impl Identifier for AggregateA {
+    fn name() -> &'static str
+    where
+        Self: Sized,
+    {
+        A_EVENTS
+    }
+}
+
+impl Identifier for AggregateB {
+    fn name() -> &'static str
+    where
+        Self: Sized,
+    {
+        B_EVENTS
+    }
+}
+
+#[async_trait]
+impl AggregateManager for AggregateA {
+    type State = ();
+    type Command = CommandA;
+    type Event = EventA;
+    type Error = CounterError;
+
+    fn event_store(&self) -> &(dyn EventStore<Self::Event, Self::Error> + Send + Sync) {
+        &self.event_store
+    }
+
+    fn apply_event(_id: &Uuid, state: Self::State, _: &StoreEvent<Self::Event>) -> Self::State {
+        // Take no action as this aggregate has no in memory state - only the projection is stateful
+        state
+    }
+
+    fn validate_command(_: &AggregateState<Self::State>, _: &Self::Command) -> Result<(), Self::Error> {
+        Ok(()) // No validation done on commands received in this aggregate
+    }
+
+    async fn do_handle_command(
+        &self,
+        aggregate_state: AggregateState<Self::State>,
+        _: Self::Command,
+    ) -> Result<AggregateState<Self::State>, Self::Error> {
+        self.persist(aggregate_state, vec![EventA {}]).await
+    }
+}
+
+#[async_trait]
+impl AggregateManager for AggregateB {
+    type State = ();
+    type Command = CommandB;
+    type Event = EventB;
+    type Error = CounterError;
+
+    fn event_store(&self) -> &(dyn EventStore<Self::Event, Self::Error> + Send + Sync) {
+        &self.event_store
+    }
+
+    fn apply_event(_id: &Uuid, state: Self::State, _: &StoreEvent<Self::Event>) -> Self::State {
+        // Take no action as this aggregate has no in memory state - only the projection is stateful
+        state
+    }
+
+    fn validate_command(_: &AggregateState<Self::State>, _: &Self::Command) -> Result<(), Self::Error> {
+        Ok(()) // No validation done on commands received in this aggregate
+    }
+
+    async fn do_handle_command(
+        &self,
+        aggregate_state: AggregateState<Self::State>,
+        _: Self::Command,
```

Comment:
I think the other examples show pattern matching on commands, it might be better to keep this one focused on just showing how to have a projector from multiple aggregates?

Reply:
Added very simple pattern match